src/harness/unittests/tsserverProjectSystem.ts

```diff
@@ -269,7 +263,7 @@ namespace ts {
         };
         readonly clearTimeout = (timeoutId: any): void => {
             if (typeof timeoutId === "number") {
-                this.callbackQueue.splice(timeoutId, 1);
+                unorderedRemoveItemAt(this.callbackQueue, timeoutId);

```

Comment:
Yes, I think this collection add callbacks in the order they are triggered, the order should be preserved here.